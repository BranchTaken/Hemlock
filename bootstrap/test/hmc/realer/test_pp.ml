open! Basis.Rudiments
open! Basis
open! Hmc
open Hmc.Realer
open Format

let test () =
  let rec fn = function
    | [] -> ()
    | x :: xs' -> begin
        printf "%a\n" pp x;
        fn xs'
      end
  in
  printf "@[<h>";
  fn [
    zero;
    one;
    inf;
    nan;
    neg zero;
    neg one;
    neg inf;
    create ~sign:Neg ~exponent:(Zint.of_uns 0) ~mantissa:(Nat.of_uns 0x1f0f);
    create ~sign:Neg ~exponent:(Zint.of_uns 42) ~mantissa:(Nat.of_uns 0x1f0f);

    (* Denormalized mantissas. *)
    create ~sign:Neg ~exponent:(Zint.of_uns 42) ~mantissa:(Nat.of_uns 0x1f0e);
    create ~sign:Neg ~exponent:(Zint.of_uns 42) ~mantissa:(Nat.of_uns 0x1f0c);
    create ~sign:Neg ~exponent:(Zint.of_uns 42) ~mantissa:(Nat.of_uns 0x1f08);
    (* Same as above, but with normalized mantissas. *)
    create ~sign:Neg ~exponent:(Zint.of_uns 42) ~mantissa:(Nat.of_uns 0xf87);
    create ~sign:Neg ~exponent:(Zint.of_uns 42) ~mantissa:(Nat.of_uns 0x7c3);
    create ~sign:Neg ~exponent:(Zint.of_uns 42) ~mantissa:(Nat.of_uns 0x3e1);
  ];
  printf "@]"

let _ = test ()
