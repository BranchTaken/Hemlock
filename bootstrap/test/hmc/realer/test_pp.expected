0x0p0
0x1p0
inf
nan
-0x0p0
-0x1p0
-inf
-0x1.f0fp0
-0x1.f0fp42
-0x1.f0ep42
-0x1.f0cp42
-0x1.f08p42
-0x1.f0ep42
-0x1.f0cp42
-0x1.f08p42
